### PR TITLE
python310Packages.treex: Enable more tests

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 183961` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.elegy</li>
    <li>python39Packages.elegy</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.treex</li>
    <li>python39Packages.treex</li>
  </ul>
</details>

**Comment by samuela** [comment]
Here's the elegy failure:
```
error: builder for '/nix/store/c8gi68lk2d1jrqnhxx934jr6f10pi977-python3.10-elegy-0.8.6.drv' failed with exit code 1;
       last 10 log lines:
       > no configure script, doing nothing
       > building
       > Executing pipBuildPhase
       > Creating a wheel...
       > WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
       > Processing /build/source
       >   Running command Preparing metadata (pyproject.toml)
       >   Preparing metadata (pyproject.toml) ... done
       > ERROR: Package 'elegy' requires a different Python: 3.10.6 not in '<3.10,>=3.7'
       >
       For full logs, run 'nix log /nix/store/c8gi68lk2d1jrqnhxx934jr6f10pi977-python3.10-elegy-0.8.6.drv'.
error: builder for '/nix/store/vm60ab0f1s34wlclrqb339b45gyf24kq-python3.9-elegy-0.8.6.drv' failed with exit code 1;
       last 10 log lines:
       > =========================== short test summary info ============================
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_basic - Attribute...
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_custom_batch_fn
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_loader_from_array
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_multi_item_ds - A...
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_multiprocessed - ...
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_prefetch - Attrib...
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_shuffled - Attrib...
       > FAILED tests/data/dataset_test.py::DataLoaderTestCase::test_worker_type - Att...
       > ===== 8 failed, 43 passed, 6 skipped, 1 deselected, 38 warnings in 15.20s ======
       For full logs, run 'nix log /nix/store/vm60ab0f1s34wlclrqb339b45gyf24kq-python3.9-elegy-0.8.6.drv'.
error: 2 dependencies of derivation '/nix/store/dn0marvqpw24584cmzlnzr5pv1ivppyl-review-shell.drv' failed to build
```
elegy is a pre-existing failure though, so we don't need to worry about that in this PR.

File: pkgs/development/python-modules/treex/default.nix [hunk 1]
```diff
@@ -31,15 +34,16 @@ buildPythonPackage rec {
   # At the time of writing (2022-03-29), rich is currently at version 11.0.0.
   # The treeo dependency is compatible with a patch, but not marked as such in
   # treex. See https://github.com/cgarciae/treex/issues/68.
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace 'rich = "^11.2.0"' 'rich = "*"' \
-      --replace 'treeo = "^0.0.10"' 'treeo = "*"' \
-      --replace 'certifi = "^2021.10.8"' 'certifi = "*"'
-  '';
+  pythonRelaxDeps = [
```
Comment (samuela) [inline pkgs/development/python-modules/treex/default.nix:37]: Cool, didn't know about this! Are there docs somewhere?

**Review by samuela — APPROVED** [review]
Thanks so much for fixing this @onny ! LGTM

File: pkgs/development/python-modules/treex/default.nix (hunk 1 above)
Comment (onny) [reply pkgs/development/python-modules/treex/default.nix:37]: I found this one https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md :)

**Pull request merged**